drivers/pds/controlplane/controlplane.go

```diff
@@ -0,0 +1,55 @@
+package controlplane
+
+import (
+	"github.com/portworx/torpedo/drivers/pds/api"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// ControlPlane PDS
+type ControlPlane struct {
+	ControlPlaneURL string
+	components      *api.Components
+}
+
+// GetRegistrationToken PDS
+func (cp *ControlPlane) GetRegistrationToken(tenantID string) (string, error) {
+	log.Info("Fetch the registration token.")
+
+	saClient := cp.components.ServiceAccount
+	serviceAccounts, _ := saClient.ListServiceAccounts(tenantID)
+	var agentWriterID string
+	for _, sa := range serviceAccounts {
+		if sa.GetName() == "Default-AgentWriter" {
+			agentWriterID = sa.GetId()
+		}
+	}
+	token, err := saClient.GetServiceAccountToken(agentWriterID)
+	if err != nil {
+		return "", err
+	}
+	return token.GetToken(), nil
+}
+
+// GetDNSZone fetches DNS zone for deployment.
+func (cp *ControlPlane) GetDNSZone(tenantID string) string {
+	tenantComp := cp.components.Tenant
+	tenant, err := tenantComp.GetTenant(tenantID)
+	if err != nil {
+		log.Panicf("Unable to fetch the tenant info.\n Error - %v", err)
```

Comment:
return the error and handle it in calling function, any clean up activity if trquired will not happen if panic is used